### PR TITLE
Vko android xiaomi splash screen patch

### PR DESCRIPTION
**Pull request by AenBleidd**
Fixes #

**Description of the Change**

**Alternate Designs**

**Release Notes**

### CODE REVIEW

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/SplashActivity.java [hunk 1]
```diff
@@ -69,6 +71,12 @@ public void onServiceConnected(ComponentName className, IBinder service) {
                 if(!monitor.boincMutexAcquired()) {
                     showNotExclusiveDialog();
                 }
+                if (BuildConfig.BUILD_TYPE.contains("xiaomi")) {
+                    mIsXiaomiSpecificFirstRun = !monitor.getXiaomiStateFile();
+                }
+                else {
+                    mIsXiaomiSpecificFirstRun = false;
+                }
```
Comment (Isira-Seneviratne) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/SplashActivity.java]:
```suggestion
                mIsXiaomiSpecificFirstRun = BuildConfig.BUILD_TYPE.contains("xiaomi") && !monitor.getXiaomiStateFile();
```

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiLicenseActivity.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * BOINC is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package edu.berkeley.boinc;
```
Comment (Isira-Seneviratne) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiLicenseActivity.kt]:
```suggestion
package edu.berkeley.boinc
```

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiLicenseActivity.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+        val binding = ActivityXiaomiLicenseBinding.inflate(layoutInflater)
+        binding.agreeButton.setOnClickListener {
+            startActivity(Intent(this@XiaomiLicenseActivity, XiaomiWelcomeActivity::class.java))
+        }
+        binding.cancelButton.setOnClickListener {
+            finish()
+        }
+
+        setContentView(binding.root)
```
Comment (Isira-Seneviratne) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiLicenseActivity.kt]:
```suggestion
        val binding = ActivityXiaomiLicenseBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.agreeButton.setOnClickListener {
            startActivity(Intent(this@XiaomiLicenseActivity, XiaomiWelcomeActivity::class.java))
        }
        binding.cancelButton.setOnClickListener {
            finish()
        }
```

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiWelcomeActivity.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * BOINC is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package edu.berkeley.boinc;
```
Comment (Isira-Seneviratne) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiWelcomeActivity.kt]:
```suggestion
package edu.berkeley.boinc
```

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiWelcomeActivity.kt [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+        val binding = ActivityXiaomiWelcomeBinding.inflate(layoutInflater)
+        binding.continueButton.setOnClickListener {
+            startActivity(Intent(this@XiaomiWelcomeActivity, SelectionListActivity::class.java))
+        }
+        binding.cancelButton.setOnClickListener {
+            finish()
+        }
+
+        setContentView(binding.root)
```
Comment (Isira-Seneviratne) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiWelcomeActivity.kt]:
```suggestion
        val binding = ActivityXiaomiWelcomeBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.continueButton.setOnClickListener {
            startActivity(Intent(this@XiaomiWelcomeActivity, SelectionListActivity::class.java))
        }
        binding.cancelButton.setOnClickListener {
            finish()
        }
```

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/SplashActivity.java (hunk 1 above)
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/SplashActivity.java]: Fixed

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiLicenseActivity.kt (hunk 2 above)
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiLicenseActivity.kt]: Fixed

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiLicenseActivity.kt (hunk 3 above)
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiLicenseActivity.kt]: Fixed

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiWelcomeActivity.kt (hunk 4 above)
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiWelcomeActivity.kt]: Fixed

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiWelcomeActivity.kt (hunk 5 above)
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/XiaomiWelcomeActivity.kt]: Fixed

**Pull request closed without merging**